### ISSUE TITLE
RelationScanner uses incorrect line separators on Windows

### ISSUE THREAD

**Issue by SolalPirelli** [epfldata/sc-public#14]
Since Windows uses a two-character line separator (`\r\n`), `RelationScanner`'s convoluted logic to read strings, which assumes the line separator to be a single `\n`, results in the last character of the last item in each line to be `\r` (i.e. "carriage return"), which results in very weird things when lines are printed if the order of items is changed.

For instance, if a row `one|1` is projected so it's `1|one`, the `1` ends with `\r`  thus when printed it'll be `|one`, as the `1` has been overwritten.

Here's a simplified `RelationScanner` that should work everywhere:

```scala
@deep
class RelationScanner(filename: String, delimiter: Char) {
  @dontLift private var current: List[String] = Nil
  @dontLift private val br: BufferedReader = new BufferedReader(new FileReader(filename))

  def next_string(): String =
    current match {
      case item :: rest => {
        current = rest
        item
      }
      case Nil => {
        current = br.readLine().split(delimiter).toList
        next_string()
      }
    }

  def hasNext() = {
    val f = br.ready()
    if (!f) br.close
    f
  }
}
```

**Comment by LPTK** [comment]
Just to let you know that we are working on solving the issue (we contacted the author of the `Scanner` class, who has a fix for Windows). In the meantime, you can of course use your implementation.
In this repository, we would like to have a `Scanner` with all the features the current one has, though, not just `hasNext` and `next_string`.

**Comment by LPTK** [comment]
The problem has been fixed in 1dbfcc360f3cf7820715c31c665de137f81dbb30, along with the final endline problem.
